CalendarApp/Event Handlers/EventHandler.h

```diff
@@ -0,0 +1,36 @@
+//
+//  EventHandler.h
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 7/19/22.
+//
+
+#import <Foundation/Foundation.h>
+#import "CalendarApp-Swift.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol EventHandler <NSObject>
+
+@required
+- (void)queryUserEventsOnDate:(NSDate *)date
+                   completion:(void(^_Nonnull)(NSMutableArray<Event *> * _Nullable events,
+                                               NSDate *date,
+                                               NSString * _Nullable error))completion;
+- (void)uploadWithEvent:(Event *)newEvent
+             completion:(void (^_Nonnull)(Event *event, NSDate *date, NSString * _Nullable error))completion;
+- (void)updateRemoteObjectWithEvent:(Event *)event
```

Comment:
`uploadEvent:completion:` is enough information to know it's remote imo. Same with the delete method.